File: cool/util/caseasm.java
@Override
public void caseACallExpr(final ACallExpr node){
final String label=this.newLabel();
        for(final PExpr pe:node.getExpr()){
        pe.apply(this);
        this.out("sw","$a0","0($sp)",this.comment("call: Push parameter",new Object[0]));
        this.out("addiu","$sp","$sp","-4");
        }
        this.out("move","$a0","$s0");
        this.out("bne","$a0","$zero",label,this.comment("call: protect from dispatch to void",new Object[0]));
        this.out("la","$a0","str_const0");
        this.out("li","$t1",node.getObjectId().getLine(),this.comment("call: line number",new Object[0]));
        this.out("jal","_dispatch_abort");
        this.out(label);
        this.out("lw","$t1","8($a0)",this.comment("call: ptr to dispatch table",new Object[0]));
final AClassDecl.Method m=this.klass.getMethodTable().get(node.getObjectId().getText());
        this.out("lw","$t1",m.getAddress(),this.comment("call: method %s is offset %d",m.name,m.offset));
        this.out("jalr","$t1");

        }

@Override
public void caseACallExpr(final ACallExpr node){
final String label=this.newLabel();
        for(final PExpr pe:node.getExpr()){
        pe.apply(this);
        this.out("sw","$a0","0($sp)",this.comment("call: Push parameter",new Object[0]));
        this.out("addiu","$sp","$sp","-4");
        }
        this.out("move","$a0","$s0");
        this.out("bne","$a0","$zero",label,this.comment("call: protect from dispatch to void",new Object[0]));
        this.out("la","$a0","str_const0");
        this.out("li","$t1",node.getObjectId().getLine(),this.comment("call: line number",new Object[0]));
        this.out("jal","_dispatch_abort");
        this.out(label);
        this.out("lw","$t1","8($a0)",this.comment("call: ptr to dispatch table",new Object[0]));
final AClassDecl.Method m=this.klass.getMethodTable().get(node.getObjectId().getText());
        this.out("lw","$t1",m.getAddress(),this.comment("call: method %s is offset %d",m.name,m.offset));
        this.out("jalr","$t1");

File: cool/util/caseasm.py
#se quito final 
from tkinter import *
def caseACallExpr (self, node):
    self.label=self.label
    for pe in node.getExpr():
        pe.apply(self)
        self.out("sw","$a0","0($sp)",self.comment("call: Push parameter",[]))
        self.out("addiu","$sp","$sp","-4")
    self.out("move","$a0","$s0")
    self.out("bne","$a0","$zero",self.label,self.comment("call: protect from dispatch to void",[]))
    self.out("la","$a0","str_const0")
    self.out("li","$t1",node.getObjectId().getLine(),self.comment("call: line number",[]))
    self.out("jal","_dispatch_abort")
    self.out(self.label)
    self.out("lw","$t1","8($a0)",self.comment("call: ptr to dispatch table",[]))
    m=self.klass.getMethodTable().get(node.getObjectId().getText())
    self.out("lw","$t1",m.getAddress(),self.comment("call: method %s is offset %d",m.name,m.offset))
    self.out("jalr","$t1")

